spec/02-integration/03-db/12-dao_hooks_spec.lua

```diff
@@ -0,0 +1,280 @@
+local helpers = require "spec.helpers"
+local hooks = require "kong.hooks"
+
+
+for _, strategy in helpers.each_strategy() do
+  describe("kong.db hooks [#" .. strategy .. "]", function()
+    local db, bp, s1, r1
+
+    lazy_setup(function()
+      bp, db = helpers.get_db_utils(strategy, {
+        "routes",
+        "services",
+      })
+
+      s1 = assert(bp.services:insert {
+        name = "s1",
+        url = "http://example.test",
+      })
+
+      r1 = assert(bp.routes:insert {
+        protocols = { "http" },
+        hosts = { "host1" },
+        service = s1,
+        name = "r1",
+      })
```

Comment:
`bp.services:insert` already does `assert`:

```suggestion
      s1 = bp.services:insert({
        name = "s1",
        url = "http://example.test",
      })

      r1 = bp.routes:insert({
        protocols = { "http" },
        hosts = { "host1" },
        service = s1,
        name = "r1",
      })
```